dpnp/backend/kernels/dpnp_krnl_mathematical.cpp

```diff
@@ -281,37 +281,62 @@ template <typename _KernelNameSpecialization1, typename _KernelNameSpecializatio
 class dpnp_trapz_c_kernel;
 
 template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
-void dpnp_trapz_c(void* array1_in, void* array2_in, void* result1, double dx, size_t array1_size, size_t array2_size)
+void dpnp_trapz_c(const void* array1_in, const void* array2_in, void* result1, double dx, size_t array1_size, size_t array2_size)
 {
```

Comment:
please, add checks if given pointers are not `nullptr`. At least for the first of them, `array1_in`.